torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h

```diff
@@ -0,0 +1,256 @@
+#pragma once
+#include <c10/util/complex.h>
+#include <torch/csrc/jit/codegen/cuda/arith.h>
+#include <torch/csrc/jit/codegen/cuda/ops/normalization.h>
+#include <torch/csrc/jit/codegen/cuda/python_frontend/fusion_definition.h>
+
+namespace nvfuser {
+struct RecordFunctor {
+  RecordFunctor(std::vector<size_t> _args, std::vector<size_t> _outputs)
+      : args(std::move(_args)), outputs(std::move(_outputs)) {}
+  virtual void operator()(FusionDefinition& fd) = 0;
+
+  std::vector<size_t> args;
+  std::vector<size_t> outputs;
+};
+
+struct InputTensorRecord : RecordFunctor {
+  InputTensorRecord(
+      std::vector<size_t> _outputs,
+      std::vector<int64_t> _symbolic_sizes,
+      std::vector<bool> _contiguous_info,
+      NvfDataType _dtype)
+      : RecordFunctor({}, std::move(_outputs)),
+        symbolic_sizes(std::move(_symbolic_sizes)),
+        contiguous_info(std::move(_contiguous_info)),
+        dtype(_dtype) {}
+  void operator()(FusionDefinition& fd) final {
+    auto tv = TensorViewBuilder()
+                  .ndims(symbolic_sizes.size())
+                  .contiguity(contiguous_info)
+                  .shape(symbolic_sizes)
+                  .dtype(dtype)
+                  .build();
+
+    fd.fusion_state.at(outputs.at(0)) = tv;
+    fd.addInput(tv);
+  }
+
+  std::vector<int64_t> symbolic_sizes;
```

Comment:
Haha, `symbolic_sizes` :wink:
I'm actually curious and think that we can already do real symbolic_sizes with existing C++ API. That's a separate topic.

Reply:
Should we also consider adding permutation order here?

Reply:
It certainly could be added, what do you have in mind?  (This might need to be a live conversation.)

Reply:
Yes, let's postpone this discussion.